domain/model/state/controllerstate.go

```diff
@@ -348,6 +348,74 @@ FROM   controller
 	return model, nil
 }
 
+// GetModelInfo returns the model associated with the provided uuid. This will
+// return a model, even if it's not activated, so it can be used to determine
+// the model's status. If the model does not exist then an error satisfying
+// [modelerrors.NotFound] will be returned.
+func (s *State) GetModelInfo(
+	ctx context.Context,
+	modelUUID coremodel.UUID,
+) (coremodel.ModelInfo, error) {
+	db, err := s.DB()
+	if err != nil {
+		return coremodel.ModelInfo{}, errors.Trace(err)
+	}
+
+	q := `
+SELECT &dbModel.*
+FROM v_model_all
+WHERE uuid = $dbModel.uuid
+`
+	model := dbModel{UUID: modelUUID.String()}
+	stmt, err := s.Prepare(q, model)
+	if err != nil {
+		return coremodel.ModelInfo{}, errors.Annotate(err, "preparing select model statement")
+	}
+
+	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		err := tx.Query(ctx, stmt, model).Get(&model)
+		if errors.Is(err, sqlair.ErrNoRows) {
+			return fmt.Errorf("%w for uuid %q", modelerrors.NotFound, modelUUID)
+		} else if err != nil {
+			return fmt.Errorf("getting model %q: %w", modelUUID, err)
+		}
+		return nil
+	})
+	if err != nil {
+		return coremodel.ModelInfo{}, errors.Trace(err)
+	}
+
+	info := coremodel.ModelInfo{
```

Comment:
I don't like adding more of this. In my patch to move agent version to the model DB, I noted that from the controller DB, you can only ever sparsely populate the core type, because agent version is not there. 

I'd rather use a new domain type. Note to @tlm.

Reply:
I'll speak to @tlm as a follow-up.

The general idea will be to move ModelInfo out of core and just hand it out directly with only the information it need.